src/windows/scanner.cc

```diff
@@ -186,8 +186,23 @@ ScanMountpoints(std::vector<drivelist::mountpoint_s> *const output) {
     ULONG number;
     drivelist::Debug("Getting device number");
     result = drivelist::volume::GetDeviceNumber(volume, &number);
-    if (FAILED(result))
+    if (FAILED(result)) {
+
+      // This error, "Invalid Function", can happen when attempting
+      // to get the device number of a virtual disk.
+      // See: https://github.com/resin-io-modules/drivelist/issues/198
+      //
+      // TODO: According to the documentation, the ERROR_INVALID_FUNCTION
+      // is the constant we want here, but in practice, that constant
+      // equals "1," and not the hexadecimal number outlined in the docs.
+      // See https://msdn.microsoft.com/en-us/library/windows/desktop/dd319335(v=vs.85).aspx
```

Comment:
That page specifically relates to "Portable Device" APIs, which we're not using here.
Seems like for the "standard Win32" [APIs](https://msdn.microsoft.com/en-us/library/cc231199.aspx) we're using, `ERROR_INVALID_FUNCTION` does indeed map to `0x1`, and it's just coincidence that `0x80070001` maps to `ERROR_INVALID_FUNCTION` in the "Portable Device" APIs.

I couldn't figure out what constant (if any) `0x80070001` maps to in the "standard Win32" APIs, but I think your talk of "Invalid Function" here (and in the commit message) is a bit of a red herring.
The nearest thing I could find was https://msdn.microsoft.com/en-us/library/cc704587.aspx but that lists nothing between `0x8004E031` and `0x80070005`, so perhaps `0x80070001` is simply an undocumented error code? :confused: 

All AFAICT, of course! :wink: